src/components/Sidebar/Tabs/DataTab.tsx

```diff
@@ -344,6 +380,42 @@ interface PillState {
 }
 type PillMap = Record<string, PillState[]>;
 
+function initializeDefaultFilter(
+  spec: GraphSpec,
+  data: GraphData,
+  columnTypes: Record<EncodingInfo['field'], EncodingInfo['type']>
+) {
+  const filters = getFilterList(spec);
+  let newSpec = Object.assign({}, spec);
```

Comment:
Just a heads up, this is a shallow duplication technique. If you change a nested object, you'll be mutating the original data.